jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/internal/jpa/weaving/ClassWeaver.java

```diff
@@ -1193,33 +1206,62 @@ public void addFetchGroupMethods(ClassDetails classDetails) {
         cv_isAttributeFetched.visitInsn(Opcodes.valueInt("IRETURN"));
         cv_isAttributeFetched.visitMaxs(0, 0);
 
-        MethodVisitor cv_checkFetched = cv.visitMethod(Opcodes.valueInt("ACC_PUBLIC"), "_persistence_checkFetched", "(Ljava/lang/String;)V", null, null);
-        cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 0);
-        cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 1);
-        cv_checkFetched.visitMethodInsn(Opcodes.valueInt("INVOKEVIRTUAL"), classDetails.getClassName(), "_persistence_isAttributeFetched", "(Ljava/lang/String;)Z", false);
-        gotoReturn = ASMFactory.createLabel();
-        cv_checkFetched.visitJumpInsn(Opcodes.valueInt("IFNE"), gotoReturn);
-        cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 0);
-        cv_checkFetched.visitTypeInsn(Opcodes.valueInt("CHECKCAST"), FETCHGROUP_TRACKER_SHORT_SIGNATURE);
-        cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 1);
-        cv_checkFetched.visitMethodInsn(Opcodes.valueInt("INVOKESTATIC"), ENTITY_MANAGER_IMPL_SHORT_SIGNATURE, "processUnfetchedAttribute", "(" + FETCHGROUP_TRACKER_SIGNATURE + "Ljava/lang/String;)V", false);
-        cv_checkFetched.visitLabel(gotoReturn);
-        cv_checkFetched.visitInsn(Opcodes.valueInt("RETURN"));
-        cv_checkFetched.visitMaxs(0, 0);
+        MethodVisitor cv_checkFetched = cv.visitMethod(Opcodes.valueInt("ACC_PUBLIC"), "_persistence_checkFetched", "(Ljava/lang/String;)V", null, null); {
+            Label l0 = ASMFactory.createLabel();
+            cv_checkFetched.visitLabel(l0);
+            // if (!this._persistence_isAttributeFetched(attributeName)) {
+            cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 0);
+            cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 1);
+            cv_checkFetched.visitMethodInsn(Opcodes.valueInt("INVOKEVIRTUAL"), classDetails.getClassName(), "_persistence_isAttributeFetched", "(Ljava/lang/String;)Z", false);
+            Label l1 = ASMFactory.createLabel();
+            cv_checkFetched.visitJumpInsn(Opcodes.valueInt("IFNE"), l1);
+            // String errorMsg = _persistence_getFetchGroup().
+            cv_checkFetched.visitVarInsn(Opcodes.valueInt("ALOAD"), 0);
+            cv_checkFetched.visitMethodInsn(Opcodes.valueInt("INVOKEVIRTUAL"), classDetails.getClassName(), "_persistence_getFetchGroup", "()Lorg/eclipse/persistence/queries/FetchGroup;", false);
```

Comment:
reuse existing constants defined within this class, such as `FETCHGROUP_SHORT_SIGNATURE` in this case